src/middlewared/middlewared/plugins/pool.py

```diff
@@ -531,6 +531,95 @@ def pool_extend(self, pool):
             pool['is_decrypted'] = True
         return pool
 
+    @accepts(
+        Int('oid'),
+        Dict(
+            'pool_attach',
+            Str('target_vdev', reuired=True),
+            Str('new_disk', reuired=True),
+            Str('passphrase'),
+        )
+    )
+    @job(lock=lambda args: f'pool_attach_{args[0]}')
+    async def attach(self, job, oid, options):
+        """
+        If the `oid` pool is passphrase GELI encrypted, `passphrase` must be specified for this operation to succeed.
+
+        `new_disk` is the devname of the new disk to be attached.
+
+        `target_vdev` is the GUID of the vdev where the disk needs to be attached. In case of STRIPED vdev, this
+        is the STRIPED disk GUID which will be converted to mirror. If `target_vdev` is mirror, it will be coverted
+        into a n-way mirror.
+        """
+        pool = await self._get_instance(oid)
+        verrors = ValidationErrors()
+        if not pool['is_decrypted']:
+            verrors.add('oid', 'Pool must be unlocked for this action.')
+            verrors.check()
+        topology = pool['topology']
+        topology_type = vdev = None
+        for i in topology:
+            if topology_type:
+                break
+            for v in topology[i]:
+                if v['guid'] == options['target_vdev']:
+                    topology_type = i
+                    vdev = v
+                    break
+        else:
+            verrors.add('pool_attach.target_vdev', 'Unable to locate VDEV')
+            verrors.check()
+        if topology_type in ('cache', 'spares'):
```

Comment:
What about other invalid types? (slog) Can we just rephrase it: `if topology_type == 'data' then process else raise` ?

Reply:
There shouldn't be any issue extending log topology ? We already allow it to be mirrored/striped, so users can add redundancy if they want to ?